### PR TITLE
modified JeongPeakWidth to include new Q_broad term 

### LINKED ISSUE #29: Implementation of Q_broad instrumental parameter
The current implementation of _r_-dependent peak broadening in the PDF due to _Q_-space resolution may contain a bug, if the literature is considered correct. This pertains to the _Q_\_broad term.

Specifically, the peak width due to the _ij_ atom pair is currently implemented as 
image1
where sigma' is the peak width due to the Debye-Waller terms, and the deltas are the standard correlated motion terms.

This is problematic, as including _Q_\_broad inside the root dictates that it _multiplies_ the intrinsic FWHM.
According to [this](https://doi.org/10.1107/S0108767391011327) reference (last line of p 342-343), the _Q_\_broad should modify the intrinsic FWHM in an _additive_ manner.
Specifically, the above reference proposes the following:
image2

In practice, the current implementation mixes instrumental and thermal sources of PDF peak broadening. This will yield different values of _Q_\_broad for different temperatures when fitting PDF data from a standard material temperature series. The values can vary by a factor of 2 or 3.  This is an issue, as the instrument resolution should not be temperature dependent.

I would propose modifying the code to define the "JeongPeakWidth" as follows:
image3

This would preserve backwards compatibility for those already using the multiplicative term, and would facilitate the use of a proper additive term as proposed in the above reference.

### PR DESCRIPTION
**Pull request by rjkoch**
…as per https://doi.org/10.1107/S0108767391011327

Should resolve #29, but may require modification of diffpy.srreal package.

@pavoljuhas @sbillinge  Ready for review

### CODE REVIEW

**Comment by rjkoch** [comment]
I neglected to update the associated header files. Should be okay now.

**Comment by rjkoch** [comment]
@pavoljuhas @sbillinge hold off on review until I can get the Travis CI build to run ok.

**Comment by rjkoch** [comment]
@pavoljuhas @sbillinge Ready for review.

File: src/diffpy/srreal/JeongPeakWidth.cpp [hunk 1]
```diff
@@ -30,14 +30,16 @@ using namespace std;
 // Constructors --------------------------------------------------------------
 
 JeongPeakWidth::JeongPeakWidth() :
-    mdelta1(0.0), mdelta2(0.0), mqbroad(0.0)
+    mdelta1(0.0), mdelta2(0.0), mqbroad(0.0), mqbroad_new(0.0)
```
Comment (sbillinge) [inline src/diffpy/srreal/JeongPeakWidth.cpp]: I think this is not a good name.  A more descriptive name would be better.  I don't know, ``separable_mqbroad`` or something along those lines?

**Comment by rjkoch** [comment]
I'm not devoted to the name as it stands. It can be refactored quite
easily. My main concern is accuracy and functionality.

What about appending _seperable instead of _new?

**Comment by sbillinge** [comment]
Yup, sounds good to me

**Comment by rjkoch** [comment]
Cool, I'll do that.

So, I can't actually get this to build on my system, mostly because I've
never built nor worked on C++ (hence I had to see Travis fail to know it
wouldn't build). Maybe there is a way for me to test it prior to merging it
in?

**Comment by sbillinge** [comment]
I don't know much about that.  Pavol or someone who knows some C would be
needed.  Does Traveis compile the c-code as well as running the tests?

**Comment by rjkoch** [comment]
It does build them in 5 environments. Maybe I can pull the built packages
from there. I did modify the tests to include the new parameter, and they
pass, but it would be good to see the PDF.

I started doing this in Pdfgui, but it's a bit more involved.

**Comment by sbillinge** [comment]
can you look at the travis build and see how it builds it and then try and
replicate that locally?  I am ok with merging it for testing, but I guess
you would still have to build it locally?  Not sure how to pull built
packages from travis tbh

**Comment by rjkoch** [comment]
Yeah, I could probably look at Travis to see how it builds. The
documentation states clearly how to build it, and I'm getting errors on
linking boost libraries, but it's not a a straight forward fix because it's
a C++ package being built by a Python module (scons)...

I don't think I would have to build it locally. When it gets installed as a
Python package through conda as part of CMI I certainly don't do any
building. It just works.

**Comment by sbillinge** [comment]
right.  The packages on conda though have gone through a full release
cycle.  This either takes a Travis built version and makes a tarball, or
else there is some compiling that Pavol does to build it before it is
released, I am not sure which.  Maybe reach out to Tim who has gone through
a full release of pdfgetx3 with instructions from Pavol, though I think
that may be pure python so we may need the same kind of help from Pavol for
this one.   Obviously we don't want to release it to conda until we are
sure it is building and running successfully....

**Comment by rjkoch** [comment]
Yeah I agree we need to test before release. I'll see about either figuring
out why my build isn't running or pulling from Travis to test. If I can't
manage I'll reach out to Tim.

**Comment by rjkoch** [comment]
I've renamed the variable as per our conversation. 

I've also learned that Travis can provide a draft release, by placing the following in the .travis.yml file:
```
deploy:
  provider: releases
  # ⋮
  draft: true
```

I've modified this file and am pushing as we speak,. Fingers crossed.

**Comment by rjkoch** [comment]
More on draft releases are here:
https://docs.travis-ci.com/user/deployment/releases/#draft-releases-with-draft-true

**Comment by sbillinge** [comment]
@rjkoch  I merged this so we can test the release etc, but we will need docs for sure.  this is a major change in functionality.  Maybe even a paper?  I am interested to see if this is more stable than the original Qbroad which I never felt was very stable.

**Comment by rjkoch** [comment]
Ok, I have beamtime this week till Thursday, but I'll work on this when I
can.

I definitely am interesting in testing it, and can start on documentation.
If it works well and is more stable, I think a paper would definitely be in
order. I don't think the science is necessarily new, as I know some codes
are already using such an implementation, but this new functionality in CMI
is definitely worth putting out there.

**Comment by sbillinge** [comment]
Yes a code release will be good for the community.  The paper would just be
comparing the two implementations since so many have used the old one. and
advertising which is better.  It should be short and sweet (and the testing
also).    I wasn't actually aware that it was implemented wrong to be
honest, it probably dates back to Thomas and PDFFIT.  Actually, strictly
there is no "wrong" or "right" because it is an empirical correction with
no scientific basis for the functional form, but there will definitely be
better and worse implementations and I think the separable one is almost
certainly going to be better, for all the reasons you gave, not to mention
the fact that I was never satisfied with the stability and reproducibility
of the old one.

All in all, worth a short paper I think, if we can find the time.

S

**Comment by rjkoch** [comment]
Did something else change this morning? Initially, things looked good, but
now the Travis page for this library (and indeed all the diffpy
libraries....) are down..?

**Comment by sbillinge** [comment]
I didn't do anything.  Let me take a look

S

**Comment by sbillinge** [comment]
This PR is merged....which is why there is no travis build.  You can ``git pull upstream master`` and run tests locally I guess?

**Pull request merged**